In [36]:
import scipy.io as sio
import numpy as np
import h5py as h5
import pandas as pd

In [2]:
mat = "../cs274c-data/imdb.mat"

For matlab arrays before v7.3, use scipy.io.loadmat('file.mat')

For matlab arrays v7.3 and above, use h5py.File('file.mat','r')

To see contents of file.mat without loading into memory, use scipy.io.whosmat('file.mat')

In [3]:
sio.whosmat(mat)

[('imdb', (1, 1), 'struct')]

In [4]:
f = sio.loadmat(mat)

In [7]:
type(f)

dict

In [20]:
for key in f.keys():
    print(key)

__header__
__version__
__globals__
imdb


In [19]:
data = f['imdb'][0][0]
print(type(data))
print(len(data))

<class 'numpy.void'>
10


In [29]:
data.dtype

dtype([('dob', 'O'), ('photo_taken', 'O'), ('full_path', 'O'), ('gender', 'O'), ('name', 'O'), ('face_location', 'O'), ('face_score', 'O'), ('second_face_score', 'O'), ('celeb_names', 'O'), ('celeb_id', 'O')])

In [69]:
for i in range(len(data)):
    print(data[i][0][5], data[i][0][5].dtype)

702986 int32
1991 uint16
['02/nm0000002_rm1075631616_1924-9-16_1991.jpg'] <U44
0.0 float64
['Lauren Bacall'] <U13
[[  686.67682059   458.46721373  1198.10043604   969.89082917]] float64
3.49056309095 float64
1.34533527716 float64
['A.D. Miles'] <U10
11516 uint16


In [46]:
print(type(data[2][0]))
data[2][0]

<class 'numpy.ndarray'>


array([ array(['01/nm0000001_rm124825600_1899-5-10_1968.jpg'],
      dtype='<U43'),
       array(['01/nm0000001_rm3343756032_1899-5-10_1970.jpg'],
      dtype='<U44'),
       array(['01/nm0000001_rm577153792_1899-5-10_1968.jpg'],
      dtype='<U43'),
       ...,
       array(['08/nm3994408_rm926592512_1989-12-29_2011.jpg'],
      dtype='<U44'),
       array(['08/nm3994408_rm943369728_1989-12-29_2011.jpg'],
      dtype='<U44'),
       array(['08/nm3994408_rm976924160_1989-12-29_2011.jpg'],
      dtype='<U44')], dtype=object)

In [44]:
print(type(data[3][0]))
data[3][0]

<class 'numpy.ndarray'>


array([ 1.,  1.,  1., ...,  0.,  0.,  0.])

In [76]:
k = 0
for i in range(len(data)):
    print(len(data[i]))
    k += 1
k

1
1
1
1
1
1
1
1
1
1


10

In [83]:
data[3][0]

array([ 1.,  1.,  1., ...,  0.,  0.,  0.])

In [84]:
d = {'filename': data[2][0], 'gender': data[3][0]}

In [86]:
df = pd.DataFrame(data=d)

In [96]:
df.filename[0][0]

'01/nm0000001_rm124825600_1899-5-10_1968.jpg'

In [97]:
df['path'] = df.filename.apply(lambda x: x[0])

In [98]:
df.head()

filename  gender  \
0   [01/nm0000001_rm124825600_1899-5-10_1968.jpg]     1.0   
1  [01/nm0000001_rm3343756032_1899-5-10_1970.jpg]     1.0   
2   [01/nm0000001_rm577153792_1899-5-10_1968.jpg]     1.0   
3   [01/nm0000001_rm946909184_1899-5-10_1968.jpg]     1.0   
4   [01/nm0000001_rm980463616_1899-5-10_1968.jpg]     1.0   

                                           path  
0   01/nm0000001_rm124825600_1899-5-10_1968.jpg  
1  01/nm0000001_rm3343756032_1899-5-10_1970.jpg  
2   01/nm0000001_rm577153792_1899-5-10_1968.jpg  
3   01/nm0000001_rm946909184_1899-5-10_1968.jpg  
4   01/nm0000001_rm980463616_1899-5-10_1968.jpg

In [84]:
values, counts = numpy.unique(data[3], return_counts=True)

In [85]:
values

array([  0.,   1.,  nan, ...,  nan,  nan,  nan])

In [86]:
counts

array([189047, 263214,      1, ...,      1,      1,      1])

There are 189047 images of females

There are 263214 images of males